packages/lodestar/src/network/network.ts

```diff
@@ -112,31 +113,44 @@ export class Libp2pNetwork extends (EventEmitter as {new (): NetworkEventEmitter
     return discv5Discovery?.discv5?.enr ?? undefined;
   }
 
+  /**
+   * Get connected peers.
+   * @param opts PeerSearchOptions
+   */
   public getPeers(opts: Partial<PeerSearchOptions> = {}): LibP2p.Peer[] {
-    const peers = Array.from(this.libp2p.peerStore.peers.values()).filter((peer) => {
-      if (opts?.connected && !this.getPeerConnection(peer.id)) {
-        return false;
-      }
-
-      this.logger.debug("Peer supported protocols", {
-        id: peer.id.toB58String(),
-        protocols: peer.protocols,
-      });
-
-      if (opts?.supportsProtocols) {
-        for (const protocol of opts.supportsProtocols) {
+    let peers: LibP2p.Peer[];
+    const peerIdStrs = Array.from(this.libp2p.connectionManager.connections.keys());
+    const peerIds = peerIdStrs
+      .map((peerIdStr) => createFromCID(peerIdStr))
+      .filter((peerId) => this.getPeerConnection(peerId));
+    peers = peerIds.map((peerId) => this.libp2p.peerStore.get(peerId)).filter(notNullish);
+
+    if (opts?.supportsProtocols) {
+      const supportsProtocols = opts.supportsProtocols;
+      peers = peers.filter((peer) => {
```

Comment:
Depending on how big the peers array is it might be worth it to join this .filter with the one above